lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -162,38 +160,38 @@ class FullPageScreenshot extends FRGatherer {
 
     // In case some other program is controlling emulation, remember what the device looks
     // like now and reset after gatherer is done.
-    let observedDeviceMetrics;
-    const lighthouseControlsEmulation = !settings.screenEmulation.disabled;
-    if (!lighthouseControlsEmulation) {
-      observedDeviceMetrics = await executionContext.evaluate(getObservedDeviceMetrics, {
-        args: [],
-        useIsolation: true,
-        deps: [kebabCaseToCamelCase],
-      });
+    const observedDeviceMetrics = await executionContext.evaluate(getObservedDeviceMetrics, {
+      args: [],
+      useIsolation: true,
+      deps: [kebabCaseToCamelCase],
+    });
+
+    // If we're gathering with mobile screenEmulation on (overlay scrollbars, etc), continue to use that for this screenshot.
+    // If screen emulation is disabled, use formFactor to determine if we are on mobile.
+    let mobile = settings.screenEmulation.mobile;
+    if (settings.screenEmulation.disabled) {
+      mobile = settings.formFactor === 'mobile';
     }
 
     try {
       return {
-        screenshot: await this._takeScreenshot(context),
+        screenshot: await this._takeScreenshot(context, observedDeviceMetrics, mobile),
         nodes: await this._resolveNodes(context),
       };
     } finally {
-      // Revert resized page.
-      if (lighthouseControlsEmulation) {
-        await emulation.emulate(session, settings);
-      } else if (observedDeviceMetrics) {
-        // Best effort to reset emulation to what it was.
-        // https://github.com/GoogleChrome/lighthouse/pull/10716#discussion_r428970681
-        // TODO: seems like this would be brittle. Should at least work for devtools, but what
-        // about scripted puppeteer usages? Better to introduce a "setEmulation" callback
-        // in the LH runner api, which for ex. puppeteer consumers would setup puppeteer emulation,
-        // and then just call that to reset?
-        // https://github.com/GoogleChrome/lighthouse/issues/11122
-        await session.sendCommand('Emulation.setDeviceMetricsOverride', {
-          mobile: settings.formFactor === 'mobile',
-          ...observedDeviceMetrics,
-        });
-      }
+      // Best effort to reset emulation to what it was.
```

Comment:
It's gonna take me awhile to validate these changes, but the first thing that jumps out to me is that the intention/comments behind the `observedDeviceMetrics` are no longer reflected in this PR. We wanted to redo the exact emulation via the `emulation` module if we controlled emulation, otherwise did a best-effort to restore what it was before. Now, it just always does the "best-effort", which is likely worse than doing it ourself if we're in control.

Reply:
I wanted to controlled which emulation variables were being changed as much as possible. We should only be changing the viewport height, and then changing it back to its original value.

In hindsight, I think I was paranoid that `emulation` would do more than reset the `height` because we give it positive values for `width` and `deviceScaleFactor` and it calls other protocol functions as well. I'm fine changing this back.